doc/user-guides/acrn-dm-parameters.rst

```diff
@@ -451,3 +454,155 @@ Here are descriptions for each of these ``acrn-dm`` command line parameters:
+
+   * - "virtio-gpio"
+     - Virtio GPIO type device, parameters format is like
+       virtio-gpio,<@controller_name{offset|name[=mapping_name]:offset|name[=mapping_name]:...}@controller_name{...}...]>
+       "controller_name", input ls /sys/bus/gpio/devices to check native gpio controller information. Usually,
+       the devices represent the controller_name, you can use it as controller_name directly. You can also
+       input cat /sys/bus/gpio/device/XXX/dev to get device id that can be used to match /dev/XXX, then use
+       XXX as the controller_name. On APL platform and Intel NUC platforms, the controller_name are gpiochip0,
+       gpiochip1, gpiochip2.gpiochip3.
+       "offset|name",you can use gpio offset or its name to locate one native gpio within the gpio controller.
+       "mapping_name",this is optional, if you want to use a customized name for a FE gpio, you can set a
+       new name for a FE virtual gpio.
+
+   * - "virtio-rnd"
+     - Virtio random generater type device, with string "kernal=on" to initialize virtio random generater state.
+
+   * - "virtio-rpmb"
+     - Virtio Replay Protected Memory Block (RPMB) type device, with "physical_rpmb" to specify RPMB in physical mode,
+       otherwise RPMB is in simulated mode.
+
+   * - "virtio-audio"
+     - Virtio audio type device
+
+   * - "virtio-net"
+     - Virtio network type device, parameter should be appended with the format
+       virtio-net,<tap_name|macvtap_name>,[vhost],[mac=<XX:XX:XX:XX:XX:XX>]
+       mac address is optional,"tap_name" needs to include the keyword tap.
+       vhost specifies vhost backend, otherwise use VBSU backend.
+
+   * - "passthru"
+     - Passthrough device, parameters below could be appended.
+       "keep_gsi", to keep vGSI for MSI capable .
+       "no_reset", do reset for passthrough PCI device by default when assigning it to post-launched VM, the parameter
+       could enable no_reset for debug only.
+       "d3hot_reset", to launch Windows post-launched VM, this parameter should be appended to enable window uefi apci bug fix.
+       "gpu", create the dedicated "igd-lpc" on 00:1f.0 for IGD passthrough.
+       "vmsix_on_msi", "vmsix_on_msi,<bar_id>" is added to specify the enablement of vMSI-X emulation based on MSI capability.
+       The specific virtual bar will be allocated.
+       "enable_ptm", enable PCIE precise time measurement mechnism for the passthrough device.
+
+   * - "uart"
+     - Emulated PCI UART, user could use the parameter with the format
+       uart,vuart_idx:<0~9> to specify hypervisor emulated pci vuart index.
+
+   * - "npk"
+     - Intel Trace Hub (known as North Peak or NPK) is a set of hardware blocks that
+       produce, switch, and output trace data from multiple hardware, firmware,
+       and software sources, used to perform full system debugging. Parameter with the format
+       npk,<master_offset>/<master number> could be appended to specify the master offset in the physical STMR
```

Comment:
STMR?

Reply:
STMR and CSR are bar region for NPK ,CSR is for config ,STMR is for master to output trace